include/nuttx/fs/fs.h

```diff
@@ -484,6 +484,15 @@ struct filelist
   spinlock_t        fl_lock;    /* Manage access to the file list */
   uint8_t           fl_rows;    /* The number of rows of fl_files array */
   FAR struct file **fl_files;   /* The pointer of two layer file descriptors array */
+
+  /* Pre-allocated files to avoid allocator access during thread creation
+   * phase, For functional safety requirements, increase
+   * CONFIG_NFILE_DESCRIPTORS_PER_BLOCK could also avoid allocator access
+   * caused by the file descriptor exceeding the limit.
+   */
+
+  FAR struct file  *fl_prefilep;
```

Comment:
why need add this field?

Reply:
pre-allocated file row for `fl_files`

Reply:
why not pass fl_prefiles directl

Reply:
Do you mean` list->fl_files = &list->fl_prefiles;` ? `fl_prefiles` is array, address of  `&fl_prefiles` with `fl_prefiles` is same.